src/main/java/athena/TaskList.java

```diff
@@ -175,8 +174,12 @@ private boolean isTimeClash(Task taskToCompare, Task task) {
 
     private boolean isIndividualTimeClash(LocalTime taskStartTime, LocalTime taskEndTime,
                                           LocalTime existingTaskStartTime, LocalTime existingTaskEndTime) {
-        return !(taskEndTime.compareTo(existingTaskStartTime) <= 0
-                || taskStartTime.compareTo(existingTaskEndTime) >= 0);
+        boolean isTimeClash = !(taskEndTime.compareTo(existingTaskStartTime) <= 0
+                || (taskStartTime.compareTo(existingTaskEndTime) >= 0
+                && existingTaskEndTime.compareTo(existingTaskStartTime) > 0));
+        boolean isMidnightClash = taskEndTime.compareTo(taskStartTime) < 0
+                && existingTaskEndTime.compareTo(existingTaskStartTime) < 0;
```

Comment:
Do we need to check the existing tasks?

Reply:
Yup for `isMidnightClash` the algo basically checks if both the end times are less than their respective start times. If so, both tasks go past 12am which means there's a clash at the 12am mark.

Reply:
Shouldn't existing tasks not go past the 12am mark? Wouldn't we have stopped that from happening when it was added/edited?

Reply:
As in ATHENA still alows task to be added provided they end at 12am but not past that